### ISSUE TITLE
private_key_jwt client authentication method needs documentation

### ISSUE THREAD

**Issue by varrunramani-okta** [okta/terraform-provider-okta#152]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform -v
Terraform v0.12.28
+ provider.okta v3.3.0
```

### Affected Resource(s)

* okta_app_oauth

### Expected Behavior
`private_key_jwt` should be documented as supported `token_endpoint_auth_method` with example

### Actual Behavior
Current docs do not reflect `private_key_jwt` as a valid type for `token_endpoint_auth_method` and `jwks` support is undocumented.
```
token_endpoint_auth_method - (Optional) Requested authentication method for the token endpoint. It can be set to "none", "client_secret_post", "client_secret_basic", "client_secret_jwt".
```

**Comment by tannernicol-okta** [comment]
👍

**Comment by noinarisak** [comment]
Closed #151 🌮
